### PR TITLE
new(userspace/libsinsp): expose threadinfo cgroups in plugins table api

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area libsinsp
/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This is needed to ensure k8smeta plugin can improve its perf.
Also, it is mandatory for the possible container engine as a plugin refactor because we will need to access thread cgroups from the plugin.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Most of the lines changes come from the new test. Most of real code is under table_adapters.h for the new `std::pair` adapter.

**Does this PR introduce a user-facing change?**:

```release-note
new(userspace/libsinsp): expose threadinfo cgroups in plugins table api
```

### CODE REVIEW

File: userspace/libsinsp/state/table_adapters.h [hunk 1]
```diff
@@ -47,6 +47,92 @@ class fixed_dynamic_fields_infos : public dynamic_struct::field_infos {
...
+	// note: this dynamic definitions are fixed in size and structure,
+	// so there's no need of worrying about specific identifier checks
+	// as they should be safely interchangeable
+	static const constexpr uintptr_t s_dynamic_fields_id = 4321;
+
+	struct dynamic_fields_t : public fixed_dynamic_fields_infos {
+		using _dfi = dynamic_struct::field_info;
+
+		inline dynamic_fields_t():
+		        fixed_dynamic_fields_infos(
```
Comment (FedeDP) [inline userspace/libsinsp/state/table_adapters.h:70]: Fields will be called `pair.first` and `pair.second`.

File: userspace/libsinsp/threadinfo.h [hunk 2]
```diff
@@ -443,10 +443,10 @@ class SINSP_PUBLIC sinsp_threadinfo : public libsinsp::state::table_entry {
 	bool m_exe_lower_layer;  ///< True if the executable file belongs to lower layer in overlayfs
 	bool m_exe_from_memfd;   ///< True if the executable is stored in fileless memory referenced by
 	                         ///< memfd
-	std::vector<std::string> m_args;       ///< Command line arguments (e.g. "-d1")
-	std::vector<std::string> m_env;        ///< Environment variables
-	std::unique_ptr<cgroups_t> m_cgroups;  ///< subsystem-cgroup pairs
```
Comment (FedeDP) [inline userspace/libsinsp/threadinfo.h:448]: Had to revert this change...

File: userspace/libsinsp/threadinfo.h [hunk 3]
```diff
@@ -659,6 +660,10 @@ class SINSP_PUBLIC sinsp_threadinfo : public libsinsp::state::table_entry {
 	bool m_parent_loop_detected;
 	libsinsp::state::stl_container_table_adapter<decltype(m_args)> m_args_table_adapter;
 	libsinsp::state::stl_container_table_adapter<decltype(m_env)> m_env_table_adapter;
+	libsinsp::state::stl_container_table_adapter<
```
Comment (FedeDP) [inline userspace/libsinsp/threadinfo.h:663]: This is a bit ugly because we need to enforce the specific value unwrapping class (the second entry).

**Comment by FedeDP** [comment]
/milestone 0.19.0
/cc @jasondellaluce @mrgian 

for a review :) even if this is still wip, i want to gather feedback asap (eg: change field names, or anything else i got wrong)

**Comment by FedeDP** [comment]
/hold
For a review by Jason!

File: userspace/libsinsp/state/table_adapters.h (hunk 1 above)
Comment (mrgian) [reply userspace/libsinsp/state/table_adapters.h:70]: IMHO you should rename these fields to `first` and `second` to make them simpler :)
I see no other possible fields the pair type, and even if we want to add other fields, I see no reason why we should categorize them by putting a prefix before the name.

Comment (FedeDP) [reply userspace/libsinsp/state/table_adapters.h:70]: I don't like the fact that we have a generic `value` and then we would have `first` and `second`.
Another solution would be `value.first` and `value.second`; indeed, i like the categorization, from an UX perspective it seems easier to guess for the developer.
I have no strong opinions though, can easily change it if you all agree on just `first` and `second` :D @jasondellaluce

**Comment by FedeDP** [comment]
Also, should i bump plugin api version?

**Review by jasondellaluce — APPROVED** [review]

**Review by LucaGuerra — APPROVED** [review]
